server/plugin.go

```diff
@@ -302,6 +302,10 @@ func (p *Plugin) OnActivate() error {
 
 		// run metrics server to expose data
 		p.runMetricsServer()
+	} else {
+		// Give metricsService a concrete, but nil value, making the interface itself
+		// not-nil and safe to use without `metrics != nil` checks everywhere.
+		p.metricsService = metrics.NilMetrics()
```

Comment:
This is the other important part of this change. Another option would actually be to always initialize `metrics` but just not expose the port. Thoughts?

Reply:
Always providing a non-nil Metrics would be safest as things are changed/maintained by future developers, and easier to reason about.